### PR TITLE
Move Nullable into core

### PR DESCRIPTION
**Pull request by dakrone**
This moves the `Nullable` annotation into the elasticsearch-core project, so it
may be used without relying entirely on the server jar. This will allow us to
decouple more pieces to make them smaller.

In addition, there were two different `Nullable` annotations, these have all
been moved to the ES version rather than the inject version.

### CODE REVIEW

**Review by rjernst — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/transport/ConnectionProfile.java [hunk 1]
```diff
@@ -41,7 +41,7 @@
      */
     public static ConnectionProfile buildSingleChannelProfile(TransportRequestOptions.Type channelType,
                                                               @Nullable TimeValue connectTimeout,
-                                                               @Nullable TimeValue handshakeTimeout) {
+                                                              @Nullable TimeValue handshakeTimeout) {
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/transport/ConnectionProfile.java]: I can't unsee this, it is like 🔥 in my 👀.

**Review by jasontedor — APPROVED** [review]
LGTM.

File: server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/status/SnapshotStatus.java [hunk 2]
```diff
@@ -22,7 +22,7 @@
 import org.elasticsearch.Version;
 import org.elasticsearch.cluster.SnapshotsInProgress.State;
 import org.elasticsearch.common.Strings;
-import org.elasticsearch.common.inject.internal.Nullable;
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/status/SnapshotStatus.java]: :heart:

**Review by nik9000 — APPROVED** [review]
You get all the approves.

**Comment by dakrone** [comment]
Thanks everyone :)

**Pull request merged**